kura/org.eclipse.kura.wire.component.provider/src/main/java/org/eclipse/kura/internal/wire/common/DbServiceHelper.java

```diff
@@ -76,7 +76,6 @@ public static DbServiceHelper getInstance(final DbService dbService) {
      *             if argument is null
      */
     public void close(final Connection conn) {
-        requireNonNull(conn, message.connectionNonNull());
```

Comment:
According to my opinion, accepting a null is a violation of contract and it does not symbolize the nature of true object orientation. That's why such invariant is very helpful in implementing methods. I agree that `DbService.close()` method also accepts null but does not work on it because of its null check but I personally don't feel it's a good practice when it's a public API.